cfgrammar/src/lib/yacc/ast.rs

```diff
@@ -5,10 +5,56 @@ use std::{
 
 use indexmap::{IndexMap, IndexSet};
 
-use super::{Precedence, YaccGrammarError, YaccGrammarErrorKind};
+use super::{parser::YaccParser, Precedence, YaccGrammarError, YaccGrammarErrorKind, YaccKind};
 
 use crate::Span;
 
+pub struct ASTValidation {
+    pub ast: GrammarAST,
+    pub errs: Vec<YaccGrammarError>,
+}
+
+/// Provides access to an ast which may or may not be valid,
+/// An AST which is invalid is provided for further analysis.
+impl ASTValidation {
+    /// The AST will be invalid if errors were encountered during
+    /// construction in which case `errors()` will be non-empty.
+    pub fn new(yacc_kind: YaccKind, s: &str) -> Self {
+        let mut errs = Vec::new();
+        let ast = match yacc_kind {
+            YaccKind::Original(_) | YaccKind::Grmtools | YaccKind::Eco => {
+                let mut yp = YaccParser::new(yacc_kind, s.to_string());
+                let _ = yp.parse().map_err(|e| errs.extend(e));
```

Comment:
Is this equivalent to `yp.parse().map_err(|e| errs.extend(e)).ok()` or similar?

Reply:
Yeah, that seems clearer.

Reply:
Should be fixed in [1e6c7d3](https://github.com/softdevteam/grmtools/pull/354/commits/1e6c7d3556741f4c9dbad09dce4b5bef2f28e5b8)